overlord/devicestate/firstboot.go

```diff
@@ -223,7 +223,16 @@ func populateStateFromSeedImpl(st *state.State, opts *populateStateFromSeedOptio
 	}
 
 	for _, seedSnap := range essentialSeedSnaps {
-		ts, info, err := installSeedSnap(st, seedSnap, snapstate.Flags{SkipConfigure: true})
+		flags := snapstate.Flags{
+			SkipConfigure: true,
+			// for dangerous models, allow all devmode snaps
+			// XXX: eventually we may need to allow specific snaps to be devmode for
+			// non-dangerous models, we can do that here since that information will
+			// probably be in the model assertion which we have here
+			ApplySnapDevMode: model.Grade() == asserts.ModelDangerous,
```

Comment:
@anonymouse64 This bit is not tested because this loop is about snapd, base, kernel, and gadget. It is probably only relevant for gadgets fwiw. We can address it maybe in a follow-up I suppose.

Reply:
I'll work on adding more tests tomorrow